usl_models/trainer/flood_task.py

```diff
@@ -0,0 +1,133 @@
+    type=str,
+    default="single",
+    help="distributed training strategy",
+    choices=["single", "mirrored", "multiworker", "tpu"],
+)
+
+args = parser.parse_args()
+
+logging.info("DEVICES" + str(device_lib.list_local_devices()))
+
+# Single Machine, single compute device
+if args.distribute == "single":
+    if tf.test.is_gpu_available():
+        strategy = tf.distribute.OneDeviceStrategy(device="/gpu:0")
+    else:
+        strategy = tf.distribute.OneDeviceStrategy(device="/cpu:0")
+    logging.info("Single device training")
+# Single Machine, multiple compute device
+elif args.distribute == "mirrored":
+    strategy = tf.distribute.MirroredStrategy()
+    logging.info("Mirrored Strategy distributed training")
+# Multi Machine, multiple compute device
+elif args.distribute == "multiworker":
+    strategy = tf.distribute.MultiWorkerMirroredStrategy()
+    logging.info("Multi-worker Strategy distributed training")
+    logging.info("TF_CONFIG = {}".format(os.environ.get("TF_CONFIG", "Not found")))
+    # Single Machine, multiple TPU devices
+elif args.distribute == "tpu":
+    cluster_resolver = tf.distribute.cluster_resolver.TPUClusterResolver(tpu="local")
+    tf.config.experimental_connect_to_cluster(cluster_resolver)
+    tf.tpu.experimental.initialize_tpu_system(cluster_resolver)
+    strategy = tf.distribute.TPUStrategy(cluster_resolver)
+    print("All devices: ", tf.config.list_logical_devices("TPU"))
+
+logging.info("num_replicas_in_sync = {}".format(strategy.num_replicas_in_sync))
+
+
+def _is_chief(task_type, task_id):
+    """Checks for primary if multiworker training."""
+    return (
+        (task_type == "chief")
+        or (task_type == "worker" and task_id == 0)
+        or task_type is None
+    )
+
+
+def train(model, dataset):
+    kwargs = {}
+    if args.epochs is not None:
+        kwargs["epochs"] = args.epochs
+
+    model.fit(dataset, **kwargs)
+
+    if args.distribute == "multiworker":
+        task_type, task_id = (
+            strategy.cluster_resolver.task_type,
+            strategy.cluster_resolver.task_id,
+        )
+    else:
+        task_type, task_id = None, None
+
+    if args.distribute == "tpu":
+        save_locally = tf.saved_model.SaveOptions(
+            experimental_io_device="/job:localhost"
+        )
+        logging.info("Saving model to %s", args.model_dir)
+        model.save_model(args.model_dir, options=save_locally)
+    # single, mirrored or primary for multiworker
+    elif _is_chief(task_type, task_id):
+        logging.info("Saving model to %s", args.model_dir)
+        model.save_model(args.model_dir)
+    # non-primary workers for multi-workers
+    else:
+        # each worker saves their model instance to a unique temp location
+        worker_dir = args.model_dir + "/workertemp_" + str(task_id)
+        logging.info("Saving model to %s", worker_dir)
+        tf.io.gfile.makedirs(worker_dir)
+        model.save_model(worker_dir)
+
+
+with strategy.scope():
```

Comment:
nit: should this be guarded behind __name__ == "__main__"?

Reply:
I don't think so -- this isn't running as a script. I think it's getting imported by Vertex stuff which causes it to be run rather than run as a script -- the interface has you passing a module name